packages/backend-function/src/function_env_translator.ts

```diff
@@ -35,7 +36,14 @@ export class FunctionEnvironmentTranslator {
           `${this.amplifySsmEnvConfigKey} is a reserved environment variable name`
         );
       }
-      if (typeof value !== 'string') {
+      if (typeof value === 'undefined') {
+        throw new AmplifyUserError('InvalidFunctionConfiguration', {
+          message: `The value of environment variable ${key} is undefined.`,
+          resolution: `Ensure that all defineFunction environment variables are defined.`,
+        });
```

Comment:
I wonder if we should be silent instead?

Having an optional boolean-ish env var (like feature flag) is quite common use case.

Reply:
That's what I originally implemented, but I thought that might hide mistakes that could have been caught at build time and instead they are caught at runtime when the function doesn't have an expected env var.

Reply:
I'm open to going with what you suggested, but if we do we should make `undefined` an explicitly allowed input so it's clear we aren't going to try to validate that case.

Reply:
What I had in mind was implicitly undefined case. I.e. variable that's not declared at all. 
If that works as expected then I'm fine with current version.

Reply:
You could do it but it would look a little wonky. It'd be something like
```ts
defineFunction({
  environment: {
    ...(process.env.THING ? { THING: process.env.THING } : {}),
  }
})

```